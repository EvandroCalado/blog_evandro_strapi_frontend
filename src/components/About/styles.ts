import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    margin: 0 auto;
    padding: 0 ${theme.spacings.md};
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    gap: ${theme.spacings.xl};

    ${Title} {
      font-size: ${theme.font.sizes['4xl']};
    }
`}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.md};

    &::first-letter {
      font-size: ${theme.font.sizes.xl};
      font-weight: bold;
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`

    & > img {
      width: 35rem;
      height: 35rem;
      overflow: hidden;
      border-radius: ${theme.spacings.md};
      object-fit: cover;
    }
`}
`;
