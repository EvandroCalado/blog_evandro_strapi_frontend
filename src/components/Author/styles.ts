import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    display: flex;
    align-items: center;
    margin: ${theme.spacings.xl} auto;

    @media screen and ${theme.media.lteOrEqSmall} {
      flex-direction: column;
      width: 30rem;
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 22rem;
    height: 17rem;
    border-radius: ${theme.spacings.sm};
    overflow: hidden;
    margin-right: ${theme.spacings.md};

    & img {
      object-fit: cover;
    }

    @media screen and ${theme.media.lteOrEqSmall} {
      width: 15rem;
      margin-bottom: ${theme.spacings.lg};
    }
`}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;

    @media screen and ${theme.media.lteOrEqSmall} {
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
    }
`}
`;

export const Content = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.sm} 0;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 3;
`}
`;
