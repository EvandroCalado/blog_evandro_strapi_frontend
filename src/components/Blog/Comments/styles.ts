import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    margin: 0 auto;

    iframe {
      color: ${theme.colors.text};
    }
`}
`;
