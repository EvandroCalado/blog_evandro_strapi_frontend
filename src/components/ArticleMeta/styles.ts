import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['7xl']};
    margin: auto;
    padding: 0 ${theme.spacings.md};
    display: flex;
    align-items: center;
    flex-direction: column;
    gap: ${theme.spacings.xl};

    & > span > a {
      font-weight: bold;
      transition: ${theme.transitions.faster};
      margin-left: ${theme.spacings.sm};

      &:hover {
        text-decoration: underline;
      }
    }

    ${Title} {
      width: 100%;
      max-width: ${theme.frameSizes['4xl']};
      text-align: center;
    }
`}
`;

export const Image = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 40rem;
    object-fit: cover;
    border-radius: ${theme.spacings.sm};
    overflow: hidden;
  `}
`;

export const ArticleContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    display: flex;
    align-items: center;
    justify-content: space-between;

    @media screen and ${theme.media.lteOrEqSmall} {
      flex-direction: column;
    }
`}
`;

export const DateContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: end;
    justify-content: center;
    gap: ${theme.spacings.lg};
`}
`;
