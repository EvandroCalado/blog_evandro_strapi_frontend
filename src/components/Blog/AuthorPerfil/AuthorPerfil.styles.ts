import styled, { css } from 'styled-components';
import { Title } from '../../Shared/Heading/Heading.styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    margin: ${theme.spacings.xl} auto;
    padding: 0 ${theme.spacings.md};
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;
    gap: ${theme.spacings.lg};


    ${Title} {
      font-size: ${theme.font.sizes['4xl']};

      @media screen and ${theme.media.lteOrEqSmall} {
        font-size: ${theme.font.sizes.xl};
      }
    }
`}
`;

export const ImageContaner = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 30rem;
    height: 30rem;
    border-radius: ${theme.spacings.sm};
    overflow: hidden;

    @media screen and ${theme.media.lteOrEqMedium} {
      width: 25rem;
      height: 25rem;
    }

    @media screen and ${theme.media.lteOrEqSmall} {
      width: 20rem;
      height: 20rem;
    }
`}
`;

export const Span = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.md};
    margin-top: ${theme.spacings.md};
`}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.md};

    @media screen and ${theme.media.lteOrEqSmall} {
      font-size: 16px;
    }
`}
`;
