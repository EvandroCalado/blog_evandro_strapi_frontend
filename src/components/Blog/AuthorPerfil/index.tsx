import Image from 'next/image';
import { Pagination, Author as StrapiAuthor } from '../../../types/strapi';
import Heading from '../../Shared/Heading';
import * as Styled from './styles';

export interface AuthorPerfilProps {
  author: StrapiAuthor;
  pagination: Pagination;
}

export default function AuthorPerfil({
  author,
  pagination,
}: AuthorPerfilProps) {
  const { image, displayName, description } = author.data.attributes;
  const { url, name } = image.data.attributes;

  return (
    <Styled.Container>
      <Styled.ImageContaner>
        <Image
          src={url}
          alt={name}
          fill
          sizes="(max-width: 768px) 100vw,
          (max-width: 1200px) 50vw,
          33vw"
          priority={true}
        />
      </Styled.ImageContaner>
      <Styled.Span>Postagens: {pagination.total}</Styled.Span>
      <Heading>{displayName}</Heading>
      <Styled.Description>{description}</Styled.Description>
    </Styled.Container>
  );
}
