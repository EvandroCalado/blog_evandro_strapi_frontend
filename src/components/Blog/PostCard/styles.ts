import styled, { css } from 'styled-components';
import { Title } from '../../Shared/Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    flex-direction: column;
    padding: ${theme.spacings.lg} ${theme.spacings.md};

    & > a {
      width: 100%;
      height: 20rem;
      margin-bottom: ${theme.spacings.md};
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 100%;
    margin-bottom: ${theme.spacings.md};

    & img {
      object-fit: cover;
      border-radius: ${theme.spacings.sm};
    }
`}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.md};

    & > a {
      transition: ${theme.transitions.faster};

      &:hover {
        text-decoration: underline;
        filter: invert(.2);
      }
    }

    ${Title} {
      overflow: hidden;
      display: -webkit-box;
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 2;
    }
`}
`;

export const Content = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.xs} 0;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 4;
`}
`;

export const DateContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
`;

export const Date = styled.span`
`;
