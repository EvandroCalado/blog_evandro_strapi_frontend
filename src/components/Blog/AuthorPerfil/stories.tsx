import { Meta, StoryFn } from '@storybook/react';
import AuthorPerfil, { AuthorPerfilProps } from '.';

import mock from '../../../utils/db.json';
const author = mock.data.posts.data[0].attributes.author;
const pagination = mock.data.posts.meta.pagination;

export default {
  title: 'Components/Blog/AuthorPerfil',
  component: AuthorPerfil,
  args: {
    author,
    pagination,
  },
  parameters: {
    layout: 'fullscreen',
  },
} as Meta;

export const Default: StoryFn<AuthorPerfilProps> = (args) => (
  <AuthorPerfil {...args} />
);
