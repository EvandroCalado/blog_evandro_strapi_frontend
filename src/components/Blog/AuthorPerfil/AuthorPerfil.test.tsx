import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../../styles/render-theme';
import AuthorPerfil from './AuthorPerfil';

import mock from '../../../utils/db.json';
const author = mock.data.posts.data[0].attributes.author;
const pagination = mock.data.posts.meta.pagination;

describe('<AuthorPerfil />', () => {
  it('should render with default values', () => {
    renderTheme(<AuthorPerfil author={author} pagination={pagination} />);
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(
      screen.getByText(`Postagens: ${pagination.total}`),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: author.data.attributes.displayName }),
    ).toBeInTheDocument();
    expect(screen.getByText(/Desenvolvedor Frontend/i)).toBeInTheDocument();
  });

  it('should render a snapshot', () => {
    const { container } = renderTheme(
      <AuthorPerfil author={author} pagination={pagination} />,
    );
    expect(container).toMatchSnapshot();
  });
});
