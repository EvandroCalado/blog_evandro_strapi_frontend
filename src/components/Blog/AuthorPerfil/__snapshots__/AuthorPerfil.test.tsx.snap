// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorPerfil /> should render a snapshot 1`] = `
.c4 {
  text-transform: none;
}

.c0 {
  width: 100%;
  max-width: 80rem;
  margin: 4rem auto;
  padding: 0 2rem;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  gap: 3rem;
}

.c0 .c3 {
  font-size: 6.4rem;
}

.c1 {
  position: relative;
  width: 30rem;
  height: 30rem;
  border-radius: 1rem;
  overflow: hidden;
}

.c2 {
  font-size: 2.4rem;
  margin-top: 2rem;
}

.c5 {
  font-size: 2.4rem;
}

@media screen and (max-width: 576px) {
  .c0 .c3 {
    font-size: 4rem;
  }
}

@media screen and (max-width: 768px) {
  .c1 {
    width: 25rem;
    height: 25rem;
  }
}

@media screen and (max-width: 576px) {
  .c1 {
    width: 20rem;
    height: 20rem;
  }
}

@media screen and (max-width: 576px) {
  .c5 {
    font-size: 16px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="110628201.jpeg"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
        src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694217416%2F110628201_8f98b88d6a.jpg&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694217416%2F110628201_8f98b88d6a.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694217416%2F110628201_8f98b88d6a.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694217416%2F110628201_8f98b88d6a.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694217416%2F110628201_8f98b88d6a.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694217416%2F110628201_8f98b88d6a.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694217416%2F110628201_8f98b88d6a.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694217416%2F110628201_8f98b88d6a.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694217416%2F110628201_8f98b88d6a.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694217416%2F110628201_8f98b88d6a.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694217416%2F110628201_8f98b88d6a.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <span
      class="c2"
    >
      Postagens: 
      4
    </span>
    <h1
      class="c3 c4"
    >
      Evandro Calado
    </h1>
    <p
      class="c5"
    >
      🚀 Desenvolvedor Frontend em constante busca por novos horizontes e desafios digitais. Meu dia a dia é uma jornada empolgante de aprendizado contínuo, onde o código é a minha tela em branco e a web é o meu playground criativo. Além de mergulhar nas linhas de código, sou abençoado por ser pai de uma menininha incrível, que me inspira a ser o meu melhor a cada dia. E quando não estou transformando ideias em experiências digitais, você pode me encontrar dando vida a ilustrações e rabiscos. Vem comigo nesta aventura em busca da excelência na criação digital !  ✨👨‍💻👨‍🎨
    </p>
  </div>
</div>
`;
